### DISCUSSION TITLE
Is is possible to do expansion when exporting variables?

### DISCUSSION THREAD

**Discussion by vext01** [magicant/yash#39] [Q&A] [answered]
I have no idea if POSIX says anything about this, but I notice a discrepancy between yash and other shells:
# bash

```
$ export A=~
$ echo $A
/home/edd
```

(similarly for `zsh`` and `pdksh`)

# yash

```
$ export A=~
$ echo $A
~
```

I noticed this when adding things to the path. If `${PATH}` contains an unexpanded `~` then it the path isn't searched as one may expect.

Is yash doing the right thing? Can it be made to behave like other shells?

Thanks.

**Comment by magicant** [comment, marked as answer]
The current POSIX does not require the bash behavior, but it will in the future. This feature is tracked in #20.

For now, you can use a separate assignment to make `~` expand:

```sh
A=~
export A
echo $A
```

**Reply by vext01** [reply]
Thanks.
